packages/pluggableWidgets/accordion-web/src/components/Accordion.tsx

```diff
@@ -21,11 +35,45 @@ export interface AccordionProps extends Pick<AccordionContainerProps, "class" |
 export function Accordion(props: AccordionProps): ReactElement | null {
     const reducer = useRef(getCollapsedAccordionGroupsReducer(props.singleExpandedGroup ? "single" : "multiple")); // the accordion group reducer function doesn't need to change during the lifetime of this component, since the singleExpandedGroup won't change.
 
-    const [collapsedAccordionGroups, collapsedAccordionGroupsDispatch] = useReducer(
+    const [accordionGroupCollapsedState, accordionGroupCollapsedStateDispatch] = useReducer(
         reducer.current,
-        props.groups.map(() => !props.previewMode && props.collapsible)
+        undefined,
+        () => {
+            const groupCollapsedStates = props.groups.map(
+                group => !props.previewMode && props.collapsible && !!group.initiallyCollapsed
+            );
+
+            if (!props.previewMode && props.singleExpandedGroup) {
+                const lastGroupCollapsedStateIndex = groupCollapsedStates.lastIndexOf(false);
+
+                if (lastGroupCollapsedStateIndex > -1) {
+                    for (let i = 0; i < lastGroupCollapsedStateIndex; i++) {
+                        groupCollapsedStates[i] = true;
+                    }
+                }
+            }
+
+            return groupCollapsedStates;
+        }
     );
 
+    const previousGroupCollapsedValues = useRef(props.groups.map(group => group.collapsed));
+
+    useMemo(() => {
+        const changes: Array<{ index: number; value: boolean }> = [];
+
+        props.groups.forEach((group, index) => {
+            if (group.collapsed !== undefined && group.collapsed !== previousGroupCollapsedValues.current[index]) {
+                changes.push({ index, value: group.collapsed });
```

Comment:
Nit: Do we really need to store the index if it's the same index as the array index? 🤔

Reply:
Simplified it.